lib/hubspot/manage/application.ex

```diff
@@ -32,37 +32,64 @@ defmodule Hubspot.Manage.Application do
   add subscriptions for all the properies that are not already subscribed.
   If a single property is subblied, it will directly add the property
   """
-  @spec add_hubspot_property_subscription([String.t()]) ::
+  @spec add_hubspot_property_subscription([map()] | map()) ::
           {:ok, String.t()} | {:error, String.t()}
-  def add_hubspot_property_subscription(property_names) when is_list(property_names) do
+  def add_hubspot_property_subscription(properties) when is_list(properties) do
     # Get all current app properties
     list_app_properties()
     |> case do
-      {:ok, app_properties} ->
-        property_names
-        |> Enum.filter(&(&1 not in app_properties))
+      {:ok,
+       %{body: %{contact_properties: contact_properties, company_properties: company_properties}}} ->
+        properties
+        |> Enum.filter(
+          &(&1["id"] not in contact_properties and &1["id"] not in company_properties)
+        )
         |> Enum.map(&add_hubspot_property_subscription/1)
+        |> Enum.all?(fn
+          {:ok, %{status: 201}} ->
+            true
+
+          {status, body} ->
+            Logger.warn(
+              "error adding property subscription with response #{status} and body #{inspect(body)}"
+            )
+
+            false
+        end)
+        |> case do
+          true -> {:ok, "property subscription added"}
+          false -> {:error, "error adding property subscription"}
+        end
```

Comment:
Looks like a good candidate for an [Enum.reduce_while/3](https://hexdocs.pm/elixir/1.12/Enum.html#reduce_while/3)